File: library/src/main/java/com/shine56/richtextx/view/RichTextView.java
package com.shine56.richtextx.view;


import android.content.Context;
import android.text.Html;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatEditText;

import com.shine56.richtextx.util.CoroutineUtil;
import com.shine56.richtextx.util.RichTextXMovementMethod;
import com.shine56.richtextx.api.HtmlTextX;
import com.shine56.richtextx.image.api.ImageBuilder;
import com.shine56.richtextx.image.ImageBuilderImpl;
import com.shine56.richtextx.image.Image;
import com.shine56.richtextx.util.RtTagHandler;

public class RichTextView extends AppCompatEditText implements HtmlTextX {

    /**
     * 构建富文本图片属性
     */
    private ImageBuilder imageBuilder;

    /**
     * 初始化
     */
    private void init(){
        setMovementMethod(RichTextXMovementMethod.Companion.getINSTANCE());

        setShowSoftInputOnFocus(false);
        setCursorVisible(false);
        setBackground(null);
        setLongClickable(false);

        imageBuilder = new ImageBuilderImpl();
    }

    @Override
    public void setTextFromHtml(String htmlText, Image image) {
        String customText;
        RtTagHandler tagHandler = new RtTagHandler(image, this, false);

        customText = htmlText.replace("span", mySpan);
        customText = customText.replace("img", myImg);

        setText(Html.fromHtml(customText, null, tagHandler));
    }

    @Override
    public ImageBuilder getImageBuilder() {
        return imageBuilder;
    }

    @Override
    public String getHtmlText() {
        return Html.toHtml(getEditableText());
    }

    public RichTextView(Context context) {
        super(context);
        init();
    }

    public RichTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public RichTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    @Override
    protected void onDetachedFromWindow() {
        CoroutineUtil.INSTANCE.cancel(hashCode());
        super.onDetachedFromWindow();
    }
}

File: library/src/main/java/com/shine56/richtextx/RichTextView.kt
package com.shine56.richtextx

import android.content.Context
import android.text.Html
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.widget.AppCompatTextView
import com.shine56.richtextx.api.HtmlTextX
import com.shine56.richtextx.api.ImageClick
import com.shine56.richtextx.api.ImageDelete
import com.shine56.richtextx.api.PhotoBuilder

class RichTextView: AppCompatTextView, HtmlTextX{
    init {
        movementMethod = RichTextXMovementMethod.INSTANCE
    }

    override fun setTextFromHtml(htmlText: String, imageGetter: Html.ImageGetter?): PhotoBuilder {
        return ParseHtmlBuilder(htmlText, imageGetter)
    }

    inner class ParseHtmlBuilder(htmlText: String, imageGetter: Html.ImageGetter?): PhotoBuilder{
        private var customText: String = ""
        private val tagHandler = RichTextXTagHandler(false, imageGetter)

        init {
            customText = htmlText.replace("span", "myspan")
            customText = customText.replace("img", "myimg")
        }

        override fun setOnCLickListener(listener: ImageClick): PhotoBuilder {
            tagHandler.setOnImageCLickListener(listener)
            return this
        }

        override fun setOnCLickListener(block: (view: View, imgUrl: String) -> Unit): PhotoBuilder {
            tagHandler.setOnImageCLickListener(ImageClick(block))
            return this
        }

        override fun setOnDeleteListener(listener: ImageDelete): PhotoBuilder {
            tagHandler.setOnImageDeleteListener(listener)
            return this
        }

        override fun setOnDeleteListener(block: (view: View, imgUrl: String) -> Unit): PhotoBuilder {
            tagHandler.setOnImageDeleteListener(ImageDelete(block))
            return this
        }

        override fun apply(){
            setText(Html.fromHtml(customText, null, tagHandler))
        }
    }

    constructor(context: Context): super(context){
    }
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
    }
    constructor(context: Context, attributeSet: AttributeSet, defStyleAttr: Int): super(context, attributeSet, defStyleAttr){
    }


}